fe/fe-core/src/main/java/com/starrocks/catalog/combinator/AggStateDesc.java

```diff
@@ -0,0 +1,191 @@
+// Copyright 2021-present StarRocks, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.starrocks.catalog.combinator;
+
+import com.google.api.client.util.Lists;
+import com.google.common.base.Objects;
+import com.google.common.base.Preconditions;
+import com.google.gson.annotations.SerializedName;
+import com.starrocks.analysis.FunctionParams;
+import com.starrocks.catalog.AggregateFunction;
+import com.starrocks.catalog.Function;
+import com.starrocks.catalog.Type;
+import com.starrocks.common.AnalysisException;
+import com.starrocks.qe.ConnectContext;
+import com.starrocks.sql.analyzer.FunctionAnalyzer;
+import com.starrocks.sql.parser.NodePosition;
+import com.starrocks.thrift.TAggStateDesc;
+import com.starrocks.thrift.TFunctionVersion;
+import com.starrocks.thrift.TTypeDesc;
+import com.starrocks.thrift.TTypeNode;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * It's a wrapper for the result type with extra information for the aggregate function.
+ */
+public class AggStateDesc {
```

Comment:
Added comment to explain why AggStateDesc is needed

Reply:
Thanks for your replies.

I will fix this in the next PR.